pkgs/development/mobile/androidenv/examples/shell.nix

```diff
@@ -0,0 +1,107 @@
+{
+  pkgs ? import ../../../../.. {},
+  pkgs_i686 ? import ../../../../.. { system = "i686-linux"; },
+  config ? pkgs.config
+}:
+
+let
+  android = {
+    versions = {
+      tools = "26.1.1";
+      platformTools = "30.0.5";
+      buildTools = "30.0.2";
+      ndk = "21.3.6528147";
+      cmake = "3.10.2";
+    };
+
+    platforms = ["23" "24" "25" "26" "27" "28" "29"];
```

Comment:
In my project I had to remove every platform but `30` or it would not find `android-30`. This is the project I tested: https://github.com/home-assistant/android

Reply:
I tried it (had to change buildTools back to 29.0.2) and it seemed to work up until this point:

```
   > Android resource linking failed
     AAPT: unknown option '--no-proguard-location-reference'.
```

 I've added API 30 to the example shell.nix too FWIW.

Reply:
It looks like it may be because home-assistant doesn't pin the build-tools version in its build.gradle to be identical to the build-tools we declare in the androidenv. Using the brand new 30.0.3 and this patch, it gets up until `:app:validateSigningFullRelease FAILED` because I don't have a keystore set up.

```patch
diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index 26e4218..f076d9a 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -13,6 +13,7 @@ plugins {

 android {
     compileSdkVersion(Config.Android.compileSdk)
+    buildToolsVersion(Config.Android.buildTools)

     ndkVersion = Config.Android.ndk

diff --git a/buildSrc/src/main/kotlin/Config.kt b/buildSrc/src/main/kotlin/Config.kt
index 7213504..72c046a 100644
--- a/buildSrc/src/main/kotlin/Config.kt
+++ b/buildSrc/src/main/kotlin/Config.kt
@@ -13,6 +13,7 @@ object Config {
         const val minSdk = 21
         const val targetSdk = 30
         const val ndk = "21.3.6528147"
+        const val buildTools = "30.0.3"
     }

     object Dependency {
diff --git a/common/build.gradle.kts b/common/build.gradle.kts
index 9b0e63c..116e299 100644
--- a/common/build.gradle.kts
+++ b/common/build.gradle.kts
@@ -9,6 +9,7 @@ val homeAssistantAndroidRateLimitUrl: String by project

 android {
     compileSdkVersion(Config.Android.compileSdk)
+    buildToolsVersion(Config.Android.buildTools)

     defaultConfig {
         minSdkVersion(Config.Android.minSdk)
```

Reply:
I've added documentation on this since it's at least the second or third time I've run into this. Projects should explicitly declare a buildToolsVersion identical to what's declared in androidenv or else they seem to default to something that's probably not intended.

Reply:
Thanks for looking into this!